spec/factories/claims.rb

```diff
@@ -31,8 +31,9 @@
 
     trait :approved do
       submitted
-      approved_at { Time.zone.now }
-      approved_by { "12345" }
+      after(:create) do |claim|
+        create(:check, claim: claim)
+      end
```

Comment:
This can just be `checks { [build(:check)] }` (or `build_list` if your prefer), so you can `build` approved claims.